mpt/src/mpt.rs

```diff
@@ -0,0 +1,1668 @@
+
+        let s_mod_node_hash_rlc = meta.advice_column();
+        let c_mod_node_hash_rlc = meta.advice_column();
+
+        let acc_s = meta.advice_column();
+        let acc_mult_s = meta.advice_column();
+        let acc_c = meta.advice_column();
+        let acc_mult_c = meta.advice_column();
+
+        let sel1 = meta.advice_column();
+        let sel2 = meta.advice_column();
+
+        // NOTE: acc_mult_s and acc_mult_c wouldn't be needed if we would have
+        // big endian instead of little endian. However, then it would be much more
+        // difficult to handle the accumulator when we iterate over the row.
+        // For example, big endian would mean to compute acc = acc * mult_r + row[i],
+        // but we don't want acc to be multiplied by mult_r when row[i] = 0 where
+        // the stream already ended and 0s are only to fulfill the row.
+
+        let key_rlc = meta.advice_column();
+        let key_rlc_mult = meta.advice_column();
+        let mult_diff = meta.advice_column();
+
+        // NOTE: key_rlc_mult wouldn't be needed if we would have
+        // big endian instead of little endian. However, then it would be much more
+        // difficult to handle the accumulator when we iterate over the key.
+        // At some point (but we don't know this point at compile time), key ends
+        // and from there on the values in the row need to be 0s.
+        // However, if we are computing rlc using little endian:
+        // rlc = rlc + row[i] * acc_r,
+        // rlc will stay the same once r[i] = 0.
+        // If big endian would be used:
+        // rlc = rlc * acc_r + row[i],
+        // the rlc would be multiplied by acc_r when row[i] = 0.
+
+        let address_rlc = meta.advice_column();
+        let counter = meta.advice_column();
+        let is_storage_mod = meta.advice_column();
+        let is_nonce_mod = meta.advice_column();
+        let is_balance_mod = meta.advice_column();
+        let is_codehash_mod = meta.advice_column();
+        let is_account_delete_mod = meta.advice_column();
+        let is_non_existing_account_proof = meta.advice_column();
+        let is_non_existing_account_row = meta.advice_column();
+
+        BranchHashInParentChip::<F>::configure(
+            meta,
+            inter_start_root,
+            not_first_level,
+            q_not_first,
+            account_leaf.is_account_leaf_in_added_branch,
+            is_last_branch_child,
+            s_advices[IS_BRANCH_S_PLACEHOLDER_POS - LAYOUT_OFFSET],
+            s_advices,
+            s_mod_node_hash_rlc,
+            acc_s,
+            acc_mult_s,
+            keccak_table,
+        );
+
+        BranchHashInParentChip::<F>::configure(
+            meta,
+            inter_final_root,
+            not_first_level,
+            q_not_first,
+            account_leaf.is_account_leaf_in_added_branch,
+            is_last_branch_child,
+            s_advices[IS_BRANCH_C_PLACEHOLDER_POS - LAYOUT_OFFSET],
```

Comment:
Should it be c_advices??

Reply:
`s_advices` is ok. `s_advices` and `c_advices` in branch children rows are used for children hashes (that's why 32 columns). However, there is a special row before branch children rows, named `branch_init`. This row contains some RLP specific data, but also some other information, for example which of the children is being changed (going from `s` to `c`) and whether the branch is a regular branch or extension node. This last is used in the code above. All this additional info is stored in `s_advices` simply because there is enough space there and it is something that is the same for `s` and `c`.

Reply:
Got it. Thanks!